pkg/spoke/submarineragent/config_controller.go

```diff
@@ -243,7 +243,7 @@ func (c *submarinerConfigController) cleanupClusterEnvironment(ctx context.Conte
 	recorder events.Recorder,
 ) error {
 	switch config.Status.ManagedClusterInfo.Platform {
-	case "GCP", "Openstack":
+	case "GCP", "Openstack", "Azure":
```

Comment:
Why don't we use `isSpokePrepared` here:

```Go
if isSpokePrepared(config.Status.ManagedClusterInfo.Platform) {
    cloudProvider, err := c.cloudProviderFactory.Get(config.Status.ManagedClusterInfo, config, recorder)
   
    return ...
}

return errors.WithMessagef(c.removeAllGateways(ctx), "Failed to unlabel the gateway nodes")
```

Reply:
Done.